src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php

```diff
@@ -47,6 +43,14 @@ public function getController(Request $request)
         }
 
         if (is_array($controller)) {
+            if (isset($controller[0]) && is_string($controller[0])) {
+                $controller[0] = $this->instantiateController($controller[0]);
+            }
+
+            if (!is_callable($controller)) {
+                throw new \InvalidArgumentException(sprintf('The controller for URI "%s" is not callable. %s', $request->getPathInfo(), $this->getControllerError($controller)));
```

Comment:
this case was missing validation. otherwise it fails in the kernel with a type error at https://github.com/symfony/symfony/blob/89d1b65037172bfb8bc4de1eeb438cd5c8de80da/src/Symfony/Component/HttpKernel/Event/FilterControllerEvent.php#L32